python/lsst/meas/algorithms/convertReferenceCatalog.py

```diff
@@ -34,19 +34,262 @@
 
 import astropy
 
-from . import ConvertReferenceCatalogBase
+import lsst.afw.table
+import lsst.pipe.base
+import lsst.pex.config as pexConfig
+from lsst.daf.base import PropertyList
 
+from .indexerRegistry import IndexerRegistry
+from .readTextCatalogTask import ReadTextCatalogTask
+from . import convertRefcatManager
+from . import ReferenceObjectLoader
 
-class ConvertReferenceCatalogTask(ConvertReferenceCatalogBase):
+# The most recent Indexed Reference Catalog on-disk format version.
+LATEST_FORMAT_VERSION = 1
+
+
+def addRefCatMetadata(catalog):
+    """Add metadata to a new (not yet populated) reference catalog.
+
+    Parameters
+    ----------
+    catalog : `lsst.afw.table.SimpleCatalog`
+        Catalog to which metadata should be attached.  Will be modified
+        in-place.
+    """
+    md = catalog.getMetadata()
+    if md is None:
+        md = PropertyList()
+    md.set("REFCAT_FORMAT_VERSION", LATEST_FORMAT_VERSION)
+    catalog.setMetadata(md)
+
+
+class DatasetConfig(pexConfig.Config):
+    """Description of the on-disk storage format for the converted reference
+    catalog.
+    """
+    format_version = pexConfig.Field(
+        dtype=int,
+        doc="Version number of the persisted on-disk storage format."
+        "\nVersion 0 had Jy as flux units (default 0 for unversioned catalogs)."
+        "\nVersion 1 had nJy as flux units.",
+        default=0  # This needs to always be 0, so that unversioned catalogs are interpreted as version 0.
+    )
+    ref_dataset_name = pexConfig.Field(
+        dtype=str,
+        doc="Name of this reference catalog; this should match the name used during butler ingest.",
```

Comment:
This is required: we may want to revamp this at some point in the future, but other than this field, the only thing tracking the refcat name is the directory it's written to (and then the butler datasetType it's registered as, once it's added to a butler). I'd much rather have the name the author expected written to a config file.

`ConvertReferenceCatalogTask` is a `Task`, so it will validate the config on init (thanks, DM-971!), including when this is run on the commandline.

Reply:
Though your comment made me check on, and clean up, a couple `validate()`-related things.